backend/src/main/java/com/github/backend/service/UserService.java

```diff
@@ -1,23 +1,26 @@
 package com.github.backend.service;
 
+import com.github.backend.models.Boulder;
 import com.github.backend.models.User;
 import com.github.backend.models.UserDto;
 import com.github.backend.repo.UserRepo;
 import lombok.RequiredArgsConstructor;
 import org.springframework.security.oauth2.core.user.OAuth2User;
 import org.springframework.stereotype.Service;
+
+import java.util.List;
 import java.util.Optional;
 
 @Service
 @RequiredArgsConstructor
 public class UserService {
     private final UserRepo userRepo;
 
-    public Optional<User> getUserById(String id){
+    public Optional<User> getUserById(String id) {
```

Comment:
here you should not return an optional. return the user and use the orElseThrow() method if an user can not be found :-)

Reply:
In this case I also need the Optional. I don't want to throw an exception when user is null. If user is logging in for the first time via github, there isn't an user object in my database yet and I want to redirect the user to the SignUp Page.